src/tests/system/tests/test_identity.py

```diff
@@ -547,36 +513,141 @@ def test_identity__lookup_idmapping_of_posix_and_non_posix_user_and_group(client
 
 
 @pytest.mark.ticket(bz=1695577)
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_identity__lookup_when_auto_private_groups_is_set_to_true(client: Client, provider: GenericProvider):
+    """
+    :title: Look up users when auto private groups is set to true
+    :description:
+        When true, the user's gid will match the uid, even when the object isn't
+        a real group in the directory. If it is a real group, the user values will override
+        the name and gid. In this test, only the users exist in LDAP.
+    :setup:
+        1. Create the following users 'user_same_gid', 'user_diff_gid' and 'user_group_gid'.
+            Set the uids and gids to match the condition defined by in the username
+        2. Configure SSSD with "ldap_id_mapping = false" and "auto_private_groups = true" and start SSSD
+    :steps:
+        1. Lookup all users and compare their uid to their gid
+        2. Lookup up the user's gid
+    :expectedresults:
+        1. All users uid and gid match
+        2. All groups are found
+    :customerscenario: True
+    :requirement: IDM-SSSD-REQ: SSSD can automatically create user private groups for users
+    """
+    provider.user("user_same_gid").add(uid=111111, gid=111111)
+    provider.user("user_diff_gid").add(uid=222222, gid=333333)
+    provider.user("user_no_gid").add(uid=444444)
+
+    client.sssd.domain["auto_private_groups"] = "true"
+    client.sssd.domain["ldap_id_mapping"] = "false"
+
+    client.sssd.start()
+
+    for i in ["user_same_gid", "user_diff_gid", "user_no_gid"]:
+        assert client.tools.getent.passwd(i).uid == client.tools.getent.passwd(i).gid, "uid does not match gid!"
+        assert client.tools.getent.group(i).gid, "group is not found!"
+
```

Comment:
May I suggest to also check that the returned UID is the same that was initially set? If it happens to be a bug that, for instance, overwrites the UID with a random GID, this test will succeed.

Reply:
Will update the test.